File: 01_Programming_Basics/_04_LoopsPartOneExercise/_04_CleverLily.java
package _01_Programming_Basics._04_LoopsPartOneExercise;

import java.util.Scanner;

public class _04_CleverLily {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int age = Integer.parseInt(scan.nextLine());
        double washMachinePrice = Double.parseDouble(scan.nextLine());
        double toyPrice = Double.parseDouble(scan.nextLine());


        int toysCounter = 0;
        double sumMoney = 0;
        double money = 0;
        int count = 0;


        for (int i = 1; i <= age; i++) {
            if (i%2 !=0){
                toysCounter++;
            }else{
                money = money+10;
                sumMoney = sumMoney+money;
                count++;
            }
        }

        double allToysMoney = toyPrice * toysCounter;
        double sum = allToysMoney + sumMoney - count;
        double sum2 = sum-washMachinePrice;

        if (sum>=washMachinePrice){
            System.out.printf("Yes! %.2f", Math.abs(sum2));
        }else{
            System.out.printf("No! %.2f", Math.abs(sum2));
        }
    }
}

File: 23_C#Basics/_04_LoopsPartOneExercise/_04_CleverLily/_04_CleverLily.cs
using System;

public class _04_CleverLily
{
    public static void Main(string[] args)
    {
        // Read input values
        int age = int.Parse(Console.ReadLine());
        double washMachinePrice = double.Parse(Console.ReadLine());
        double toyPrice = double.Parse(Console.ReadLine());

        int toysCounter = 0;
        double sumMoney = 0;
        double money = 0;
        int count = 0;

        // Calculate the number of toys received and money saved for each birthday
        for (int i = 1; i <= age; i++)
        {
            if (i % 2 != 0)
            {
                toysCounter++;
            }
            else
            {
                money += 10;
                sumMoney += money;
                count++;
            }
        }

        // Calculate the total money from selling the toys and saved money
        double allToysMoney = toyPrice * toysCounter;
        double totalMoney = allToysMoney + sumMoney - count;
        double diff = Math.Abs(totalMoney - washMachinePrice);

        // Determine if Lily can buy the washing machine and print the result
        if (totalMoney >= washMachinePrice)
        {
            Console.WriteLine($"Yes! {diff:F2}");
        }
        else
        {
            Console.WriteLine($"No! {diff:F2}");
        }
    }
}
